ios/Classes/BetterPlayer.m

```diff
@@ -635,32 +635,9 @@ - (void)setVolume:(double)volume {
 }
 
 - (void)setSpeed:(double)speed result:(FlutterResult)result {
-    if (speed == 1.0 || speed == 0.0) {
-        _playerRate = 1;
-        result(nil);
-    } else if (speed < 0 || speed > 2.0) {
-        result([FlutterError errorWithCode:@"unsupported_speed"
-                                   message:@"Speed must be >= 0.0 and <= 2.0"
-                                   details:nil]);
-    } else if ((speed > 1.0 && _player.currentItem.canPlayFastForward) ||
-               (speed < 1.0 && _player.currentItem.canPlaySlowForward)) {
-        _playerRate = speed;
-        result(nil);
-    } else {
-        if (speed > 1.0) {
-            result([FlutterError errorWithCode:@"unsupported_fast_forward"
-                                       message:@"This video cannot be played fast forward"
-                                       details:nil]);
-        } else {
-            result([FlutterError errorWithCode:@"unsupported_slow_forward"
-                                       message:@"This video cannot be played slow forward"
-                                       details:nil]);
-        }
-    }
-
-    if (_isPlaying){
-        _player.rate = _playerRate;
-    }
+    _playerRate = speed;
+    _player.rate = _playerRate;
```

Comment:
``` suggestion
    if (_isPlaying) {
        _player.rate = _playerRate;
    }
```
We should keep the old check for safety (when changing the rate maybe it can affect the player's play status)
https://developer.apple.com/documentation/avfoundation/avplayer/pause()